### ISSUE TITLE
Console adds UTF-8 BOM to output

### ISSUE THREAD

**Issue by christoff-buerger** [IronScheme/IronScheme#151]
The `IronScheme.Console-v4.exe` console adds the UTF-8 BOM to output. The BOM is even emitted when `-nologo` is set.

You can test this behavior by calling
```
echo a>f
mono "$( command -v IronScheme.Console-v4.exe )" -nologo>>f
>(exit)
```
and investigating `f`.

This is very annoying, since console output not necessarily is redirected directly into a file, but could for example be processed in a pipe, which in turn now has to skip the BOM; or like in above example, output might be appended to an existing file, which now has the BOM in the middle corrupting it.

I tested this behavior with mono, but guess it is not an issue of mono, but rather the `IronScheme.Console-v4.exe` and likely `IronScheme.Console-v2.exe` implementations.

In .NET Core (version 6), the `IronScheme.ConsoleCore.exe` does not add the BOM.

**Comment by leppie** [comment]
I tested this on windows, but could not replicate. 

```
echo a>f
echo 1 | IronScheme.Console32-v2.exe >> f
echo 2 | IronScheme.Console32-v4.exe >> f
```

Maybe a mono and/or non-windows issue?

**Comment by christoff-buerger** [comment]
> I tested this on windows, but could not replicate.
> 
> ```
> echo a>f
> echo 1 | IronScheme.Console32-v2.exe >> f
> echo 2 | IronScheme.Console32-v4.exe >> f
> ```
> 
> Maybe a mono and/or non-windows issue?

@leppie : Thank you very much for your investigation and time!

I checked again and recognised, the BOM is written to stderr, not stdout! Can you double check this on Windows?

**Comment by leppie** [comment]
Tried stderr too, not seeing issues

```
echo a | IronScheme.Console32-v4.exe 2>> f
```

![image](https://github.com/IronScheme/IronScheme/assets/127666/f1638d30-24f5-444c-8ea7-11ea4582b989)

**Comment by christoff-buerger** [comment]
I guess it is a macOS, or more likely Mono issue then.

I can fool around a bit more, but the likely outcome will be that I do a workaround on my side (wrap Ironscheme calls in a script that discards the BOM).

I think you can close the issue and mark it somehow as good to know issue on Mono.

**Comment by lassik** [comment]
Possibly relevant:

* https://github.com/dotnet/runtime/issues/27258
* https://github.com/mono/mono/issues/6901

**Comment by christoff-buerger** [comment]
> Possibly relevant:
> 
>     * [System.Console unexpectedly uses a UTF-8 encoding *with BOM* on Unix dotnet/runtime#27258](https://github.com/dotnet/runtime/issues/27258)
> 
>     * [ Console.OutputEncoding = Encoding.UTF8 mono/mono#6901](https://github.com/mono/mono/issues/6901)

This looks relevant! Thanks for the good hints.

What is the implication? That `IronScheme.Console-v4.exe` should use another `Console.OutputEncoding` to fix the behavior on macOS / Unix? Or another way for getting the correct encoding (like they had with the `Encoding.GetEncoding("UTF-8")` in the first link that yields UTF-8 with BOM)?

**Comment by leppie** [comment]
Thanks for the feedback. 

The console output is set to UTF-8 in the host program.

```
System.Console.OutputEncoding = Encoding.UTF8;
```

But looking at the decompiled source, it gets this:

```
    public static Encoding UTF8
    {
        get
        {
            if (utf8Encoding == null)
            {
                utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true);
            }

            return utf8Encoding;
        }
    }    public static Encoding UTF8
    {
        get
        {
            if (utf8Encoding == null)
            {
                utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true);
            }

            return utf8Encoding;
        }
    }
```

Perhaps just setting it to `new UTF8Encoding(encoderShouldEmitUTF8Identifier: false)`  will do the trick? Not sure if `UTF8Identifier` is the BOM.

**Comment by leppie** [comment]
Fix attempt incoming

**Comment by leppie** [comment]
> Fix attempt incoming

https://github.com/IronScheme/IronScheme/releases/tag/1.0.331

@christoff-buerger see if this fixes the issue

**Comment by christoff-buerger** [comment]
> > Fix attempt incoming
> 
> https://github.com/IronScheme/IronScheme/releases/tag/1.0.331
> 
> @christoff-buerger see if this fixes the issue

@leppie: Perfect! I tested and the issue is fixed for `IronScheme.Console-v4.exe`!

I don't use `IronScheme.Console-v2.exe`; so I can't say if it ever had any BOM issue or, if so, it's fixed now.

Thank you very much!

**Comment by leppie** [comment]
Fixed in 311b8df
